app/forms/provider_interface/provider_user_invitation_wizard.rb

```diff
@@ -163,10 +163,7 @@ def previous_provider_id
     end
 
     def next_provider_needing_permissions_setup
-      providers.find do |p|
-        provider_permissions.keys.exclude?(p.to_s) ||
-          provider_permissions[p.to_s]['permissions'].reject(&:blank?).blank?
-      end
+      providers.find { |p| provider_permissions.keys.exclude?(p.to_s) }
```

Comment:
can you just do something like `(providers - provider_permissions.keys).first` or `(providers - provider_permissions.keys.map(&:to_i)).first`?